api/src/main/java/org/apache/cloudstack/api/command/user/template/CreateTemplateCmd.java

```diff
@@ -251,26 +267,33 @@ public long getEntityOwnerId() {
             }
         }
 
-        if(projectId != null){
+        Account accountToUse = callingAccount;
+        if (accountName != null && domainId != null) {
+            try {
+                accountToUse = _accountService.getAccount(_accountService.finalyzeAccountId(accountName, domainId, projectId, true));
+
+            } catch (Exception exception) {
+                s_logger.info("Unable to find accountId associated with accountName=" + accountName + " and domainId="
+                      + domainId + " using account=" + accountToUse);
+            }
+        } else if (projectId != null) {
```

Comment:
can you extract the code to (a) new method(s) @sudo87 ?

Reply:
Sure, created separate methods for access check and accountId determination